src/main/java/com/custom/fuelsdk/PaginationETSoapObject.java

```diff
@@ -0,0 +1,365 @@
+//        retrieveRequest.setObjectType(soapObjectName);
+        soap = connection.getSoap("retrieve", soapObjectName);
+      } else {
+//        retrieveRequest.setObjectType(internalType.getSimpleName());
+        soap = connection.getSoap("retrieve", internalType.getSimpleName());
+      }
+    }
+    RetrieveResponseMsg retrieveResponseMsg = soap.retrieve(retrieveRequestMsg);
+
+    if (logger.isTraceEnabled()) {
+      logger.trace("RetrieveResponseMsg:");
+      logger.trace("  requestId = " + retrieveResponseMsg.getRequestID());
+      logger.trace("  overallStatus = " + retrieveResponseMsg.getOverallStatus());
+      logger.trace("  results = {");
+      for (APIObject result : retrieveResponseMsg.getResults()) {
+        logger.trace("    " + result);
+      }
+      logger.trace("  }");
+    }
+
+    response.setRequestId(retrieveResponseMsg.getRequestID());
+    if (retrieveResponseMsg.getOverallStatus().equals("OK")) {
+      response.setStatus(ETResult.Status.OK);
+    } else if (retrieveResponseMsg.getOverallStatus().equals("Error")) {
+      response.setStatus(ETResult.Status.ERROR);
+    }
+    response.setResponseCode(retrieveResponseMsg.getOverallStatus());
+    response.setResponseMessage(retrieveResponseMsg.getOverallStatus());
+    for (APIObject internalObject : retrieveResponseMsg.getResults()) {
+      //
+      // Allocate a new (external) object:
+      //
+
+      T externalObject = null;
+      try {
+        externalObject = externalType.newInstance();
+      } catch (Exception ex) {
+        throw new ETSdkException("could not instantiate "
+                                   + externalType.getName(), ex);
+      }
+
+      externalObject.setClient(client);
+
+      //
+      // Convert from internal representation:
+      //
+
+      externalObject.fromInternal(internalObject);
+
+      //
+      // Add result to the list of results:
+      //
+
+      ETResult<T> result = new ETResult<T>();
+      result.setObject(externalObject);
+      response.addResult(result);
+    }
+
+    if (retrieveResponseMsg.getOverallStatus().equals("MoreDataAvailable")) {
+      response.setMoreResults(true);
+    }
+
+    return response;
+  }
+
+  /**
+   * @param client The ETClient object
+   * @param key    The key
+   * @return The List of Column names
+   * @throws ETSdkException
+   */
+  private static List<String> retrieveColumnNames(ETClient client,
+                                                  String key)
+    throws ETSdkException {
+    return getColumnNames(retrieveColumns(client, key));
+  }
+
+
+  /**
+   * @param columns The List of ETDataExtensionColumn object
+   * @return The List of ETDataExtension Column names
+   */
+  private static List<String> getColumnNames(List<ETDataExtensionColumn> columns) {
+    List<String> columnNames = new ArrayList<String>();
+    for (ETDataExtensionColumn column : columns) {
+      columnNames.add(column.getName());
+    }
+    return columnNames;
+  }
+
+  @Override
```

Comment:
are these empty implementations? Returning null is ok? Can you add a comment, pls